v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/range/RangePreparedStatementSetterTest.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.source.reader.io.jdbc.uniformsplitter.range;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.teleport.v2.source.reader.io.jdbc.dialectadapter.mysql.MysqlDialectAdapter;
+import com.google.cloud.teleport.v2.source.reader.io.jdbc.dialectadapter.mysql.MysqlDialectAdapter.MySqlVersion;
+import com.google.common.collect.ImmutableList;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/** Test class for {@link RangePreparedStatementSetter}. */
+@RunWith(MockitoJUnitRunner.class)
+public class RangePreparedStatementSetterTest {
```

Comment:
nit: Please name the class or methods as MySQL wherever queries are being verified.

Reply:
This test is verifying the behavior of the query produced and not exactly the query.
`MySqlDialectAdapterTest` does actually verify the query syntax (and the enclosing test class is named accordingly)

In this test(`RangePreparedStatementSetterTest`) we are testing `RangePreparedStatementSetter` (as needed by JDBCIO#readAll) for queries  on a derby table and the `RangePreparedStatementSetter` should not change across MySql or PG. 
For the UT we do use MySQl dialect adapter to get the prepared statement, currently (derby matches mySQL in a lot of aspects).